### DISCUSSION TITLE
Bug: add file contains injectHtmlTags as plugin in docusaurus.config.js ?

### DISCUSSION THREAD

**Discussion by namnguyenthanhwork** [facebook/docusaurus#10203] [Q&A] [answered]
Some prev version, i used to add file contains injectHtmlTags as plugin in docusaurus.config.js, it successfully.

<img width="1211" alt="Screenshot 2024-06-06 at 13 23 32" src="https://github.com/facebook/docusaurus/assets/78526193/74f473ec-0e40-4ad1-ae1e-428d105716fd">

But now, i can't add this. When i added and run `npm start`. Error logs show

<img width="952" alt="image" src="https://github.com/facebook/docusaurus/assets/78526193/4e7462bd-2851-4abf-8752-d043fa8c107b">

content inside plugin

```js
module.exports = function () {
  return {
    injectHtmlTags() {
      return {
        postBodyTags: [
          {
            tagName: 'script',
            innerHTML: `
            var Tawk_API=Tawk_API||{}, Tawk_LoadStart=new Date();
            (function(){
            var s1=document.createElement("script"),s0=document.getElementsByTagName("script")[0];
            s1.async=true;
            s1.src='https://embed.tawk.to/';
            s1.charset='UTF-8';
            s1.setAttribute('crossorigin','*');
            s0.parentNode.insertBefore(s1,s0);
            })();`,
          },
        ],
      };
    },
  };
};
```

Can everyone help me? Thanks.

**Comment by slorber** [comment]
This should keep working

Please provide a small repro.
What is prev version, what is now. We need to know and be able to see the error ourselves easily.

**Reply by namnguyenthanhwork** [reply]
@slorber 

This is the order that I did it in, you can test it:

1. Clone new repo docusaurus v3.4.0 `npx create-docusaurus@latest my-website classic`
2. Add new folder plugins and add file `tawkto-plugin.js` (/src/plugins/tawkto-plugin.js)
Content inside js file

```js
module.exports = function () {
  return {
    injectHtmlTags() {
      return {
        postBodyTags: [
          {
            tagName: 'script',
            innerHTML: `
            var Tawk_API=Tawk_API||{}, Tawk_LoadStart=new Date();
            (function(){
            var s1=document.createElement("script"),s0=document.getElementsByTagName("script")[0];
            s1.async=true;
            s1.src='https://embed.tawk.to/<tawkto_id>';
            s1.charset='UTF-8';
            s1.setAttribute('crossorigin','*');
            s0.parentNode.insertBefore(s1,s0);
            })();`,
          },
        ],
      };
    },
  };
};
```

In `docusaurus.config.js` add plugin

```js
 plugins: [['./src/plugins/tawkto-plugin.js', {}]],
```

3. Run command `npm start`, you will receive error log

<img width="1006" alt="image" src="https://github.com/facebook/docusaurus/assets/78526193/a2e16010-734d-4183-8be5-62bd52aa6575">

4. Downgrade docusaurus v3.4.0 -> v3.0.0 in package.json

5. Run command `npm start` -> successfully

<img width="569" alt="image" src="https://github.com/facebook/docusaurus/assets/78526193/d4911a17-d554-4381-8f7b-1790a151b9a4">

**Reply by slorber** [reply]
Can you provide a runnble repro using Docusaurus.new? I'll take a look later

It is possible this was a temporary issue, so please try again by creating a runnble repro

**Reply by namnguyenthanhwork** [reply]
@slorber You can clone repo https://github.com/namnguyenthanhwork/docusaurus-test on your local machine and test it, you will see error logs. Switch branch v3.0.0 -> run -> it's successful.

**Reply by OzakIOne** [reply]
[3.0 Repro working](https://stackblitz.com/edit/github-nq8usr-iudsrz)

[3.4 Repro error](https://stackblitz.com/edit/github-nq8usr-h19oub)

**Reply by slorber** [reply, marked as answer]
This is because the plugin does not provide a name

The name property is required, unfortunately we never ensured that, and the error only happened if you used certain lifecycle methods. 

Please add a name to that plugin. From now on we'll improve the error message, but the name will remain required. See also https://github.com/facebook/docusaurus/pull/10239
